specifyweb/specify/tree_utils.py

```diff
@@ -27,6 +27,14 @@ def get_treedef(collection: spmodels.Collection, tree_name: str) ->  List[Tuple[
     )
 
     result = list(search_query)
+
+    # Handle null taxontreedef.discipline
+    if len(result) == 0 and tree_name.lower() == 'taxon':
+        result = list(spmodels.CollectionObjectType.objects\
+            .filter(collection=collection)\
+            .annotate(item_counts=Count('taxontreedef__treedefitems'))\
+            .values_list('taxontreedef_id', 'item_counts'))
```

Comment:
Hmm should trees corresponding to COT always be searched? If yes, then this should be refactored to always include them (not just when result is empty). You can do quite a different ways of doing this. I think the "correct" way would be to modify `get_search_filters` to always return a Q() object. For taon trees, do `Q(discipline=collection.discipline) | Q(cotypes__collection=collection)`. See model definition: https://github.com/specify/specify7/blob/0557807fea85d44019919a4d9368a98065e88bbd/specifyweb/specify/models.py#L7509

Reply:
Plus, if you do this here, it'll automatically resolve a possible bug when scoping..

Reply:
Resolved in https://github.com/specify/specify7/pull/5036/commits/b62b7353e0e0ede3ee993edc46473eb1dde15cbe